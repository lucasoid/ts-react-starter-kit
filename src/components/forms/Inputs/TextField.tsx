import * as React from 'react';
import * as styles from '~components/forms/Form.css';

interface ITextFieldProps {
    value: string;
    label: string;
    onChange: Function;
    disabled?: boolean;
}

export const TextField: React.FC<ITextFieldProps> = props => {
    return (
        <div className={styles.ctrl}>
            <label>{props.label}</label>
            <input
                disabled={props.disabled}
                type="text"
                value={props.value}
                onChange={e => props.onChange(e.target.value)}
            />
        </div>
    );
};
