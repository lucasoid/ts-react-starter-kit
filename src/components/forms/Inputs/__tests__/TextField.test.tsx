import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { TextField } from '../TextField';

describe('TextField', () => {
    it('renders a styled text input', () => {
        const onChange = jest.fn();
        let wrapper = shallow(<TextField label="My test field" value="ABC" onChange={onChange} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a disabled text input', () => {
        const onChange = jest.fn();
        let wrapper = shallow(<TextField label="My test field" value="ABC" onChange={onChange} disabled={true} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('passes input changes to the onChange handler', () => {
        const onChange = jest.fn();
        let wrapper = mount(<TextField label="My test field" value="ABC" onChange={onChange} />);
        wrapper.find('input').simulate('change', { target: { value: 'DEF' } });
        expect(onChange.mock.calls[0][0]).toBe('DEF');
    });
});
