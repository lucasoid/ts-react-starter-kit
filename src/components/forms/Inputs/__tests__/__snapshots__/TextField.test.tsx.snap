// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders a disabled text input 1`] = `
<div
  className="ctrl"
>
  <label>
    My test field
  </label>
  <input
    disabled={true}
    onChange={[Function]}
    type="text"
    value="ABC"
  />
</div>
`;

exports[`TextField renders a styled text input 1`] = `
<div
  className="ctrl"
>
  <label>
    My test field
  </label>
  <input
    onChange={[Function]}
    type="text"
    value="ABC"
  />
</div>
`;
